azext_iot/tests/iothub/__init__.py

```diff
@@ -89,6 +89,27 @@ def __init__(self, test_scenario):
 
                 sleep(ROLE_ASSIGNMENT_REFRESH_TIME)
 
+                new_hub = self.cmd(
+                    "iot hub show -n {} -g {}".format(self.entity_name, self.entity_rg)
+                ).get_output_in_json()
+
+                role_assignments = self.get_role_assignments(new_hub["id"], USER_ROLE)
+                role_assignment_principal_names = [assignment["principalName"] for assignment in role_assignments]
+
+                account = self.cmd("account show").get_output_in_json()
+                user = account["user"]
+
+                if user["name"] is None:
+                    raise Exception("User not found")
+                elif user["name"] not in role_assignment_principal_names:
+                    # assign IoT Hub Data Contributor role to current user
+                    self.cmd(
+                        '''role assignment create --assignee "{}" --role "{}" --scope "{}"'''.format(
+                            user["name"], USER_ROLE, new_hub["id"]
+                        )
+                    )
+                    sleep(60)
```

Comment:
Is this an optimal application of sleep times? There is a blanket 120 second sleep time upon IoT Hub resource creation, and a single 60 second wait in case the data contributor role assignment needs to be applied. Are there opportunities to reduce blanket wait times, and/or have a periodic poll in case required role-assignments are not propagated yet?